File: airbyte-commons-worker/src/main/java/io/airbyte/workers/internal/DefaultAirbyteMessageBufferedWriter.java
/*
 * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.workers.internal;

import io.airbyte.commons.json.Jsons;
import io.airbyte.protocol.models.AirbyteMessage;
import java.io.BufferedWriter;
import java.io.IOException;

public class DefaultAirbyteMessageBufferedWriter implements AirbyteMessageBufferedWriter {

  protected final BufferedWriter writer;

  public DefaultAirbyteMessageBufferedWriter(final BufferedWriter writer) {
    this.writer = writer;
  }

  @Override
  public void write(final AirbyteMessage message) throws IOException {
    writer.write(Jsons.serialize(message));
    writer.newLine();
  }

  @Override
  public void flush() throws IOException {
    writer.flush();
  }

  @Override
  public void close() throws IOException {
    writer.close();
  }

}

File: airbyte-cdk/java/airbyte-cdk/dependencies/bin/testFixtures/io/airbyte/workers/internal/DefaultAirbyteMessageBufferedWriter.kt
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */
package io.airbyte.workers.internal

import io.airbyte.commons.json.Jsons
import io.airbyte.protocol.models.AirbyteMessage
import java.io.BufferedWriter
import java.io.IOException

class DefaultAirbyteMessageBufferedWriter(protected val writer: BufferedWriter) :
    AirbyteMessageBufferedWriter {
    @Throws(IOException::class)
    override fun write(message: AirbyteMessage) {
        writer.write(Jsons.serialize(message))
        writer.newLine()
    }

    @Throws(IOException::class)
    override fun flush() {
        writer.flush()
    }

    @Throws(IOException::class)
    override fun close() {
        writer.close()
    }
}
